File: src/com/basmilius/ps/bastools/framework/identity/liveTemplates/IdentityLiveTemplatesContext.java
package com.basmilius.ps.bastools.framework.identity.liveTemplates;

import com.basmilius.ps.bastools.framework.identity.IdentityFramework;
import com.intellij.codeInsight.template.TemplateContextType;
import com.intellij.psi.PsiFile;
import org.jetbrains.annotations.NotNull;

public class IdentityLiveTemplatesContext extends TemplateContextType
{

	/**
	 * IdentityLiveTemplatesContext Constructor.
	 */
	public IdentityLiveTemplatesContext ()
	{
		super("BASTOOLS.IDENTITY_FRAMEWORK", "Identity Framework");
	}

	/**
	 * {@inheritDoc}
	 */
	@Override
	public boolean isInContext (@NotNull final PsiFile psi, final int offset)
	{
		return psi.getName().endsWith(".php") && IdentityFramework.isIdentityFrameworkProject(psi.getProject());
	}

}

File: src/com/basmilius/ps/bastools/framework/identity/liveTemplates/IdentityLiveTemplatesContext.kt
package com.basmilius.ps.bastools.framework.identity.liveTemplates

import com.intellij.codeInsight.template.TemplateContextType
import com.intellij.psi.PsiFile
import com.intellij.util.PlatformUtils

/**
 * Class IdentityLiveTemplatesContext
 *
 * @constructor
 *
 * @author Bas Milius
 * @package com.basmilius.ps.bastools.framework.identity.liveTemplates
 */
class IdentityLiveTemplatesContext: TemplateContextType("BASTOOLS.IDENTITY_FRAMEWORK", "Identity Framework")
{

	/**
	 * {@inheritDoc}
	 *
	 * @author Bas Milius
	 */
	override fun isInContext(psi: PsiFile, offset: Int): Boolean
	{
		return PlatformUtils.isPhpStorm() && psi.name.endsWith(".php")
	}

}
